src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -163,59 +163,54 @@ if (__DEV__) {
   var debugCounter = 1;
 }
 
-// This is a constructor of a POJO instead of a constructor function for a few
-// reasons:
+function FiberNode(tag: TypeOfWork, key: null | string): Fiber {
+    // Instance
+    this.tag = tag;
+    this.key = key;
+    this.type = null;
+    this.stateNode = null;
+
+    // Fiber
+    this.return = null;
+    this.child = null;
+    this.sibling = null;
+    this.index = 0;
+
+    this.ref = null;
+
+    this.pendingProps = null;
+    this.memoizedProps = null;
+    this.updateQueue = null;
+    this.memoizedState = null;
+
+    // Effects
+    this.effectTag = NoEffect;
+    this.nextEffect = null;
+    this.firstEffect = null;
+    this.lastEffect = null;
+
+    this.pendingWorkPriority = NoWork;
+    this.progressedPriority = NoWork;
+    this.progressedChild = null;
+    this.progressedFirstDeletion = null;
+    this.progressedLastDeletion = null;
+
+    this.alternate = null;
+}
+
+// This is now a constructor function again, rather than a POJO constructor, still
+// please ensure we do the following:
 // 1) Nobody should add any instance methods on this. Instance methods can be
 //    more difficult to predict when they get optimized and they are almost
 //    never inlined properly in static compilers.
 // 2) Nobody should rely on `instanceof Fiber` for type testing. We should
 //    always know when it is a fiber.
-// 3) We can easily go from a createFiber call to calling a constructor if that
-//    is faster. The opposite is not true.
```

Comment:
The opposite is now true. :)

Maybe we can keep a comment that this should still be hidden behind the `createFiber` function so that we can change it in reverse if we need to?